src/power.jl

```diff
@@ -321,11 +321,11 @@ end
     return nothing
 end
 
-@inline function pow!(c::Taylor1{T}, a::Taylor1{T}, aux::Taylor1{T},
-        r::S, k::Int) where {T<:NumberNotSeriesN, S<:Real}
+@inline function pow!(c::Taylor1{Taylor1{T}}, a::Taylor1{Taylor1{T}}, aux::Taylor1{Taylor1{T}},
+        r::S, k::Int) where {T<:NumberNotSeries, S<:Real}
```

Comment:
What about use the limit `T<:NumberNotSeriesN`? That would allow to use this function in two and more nested Taylor1 structs...

(I think that the same comment applies to the `div!` method introduced in #387.)